src/avram/associations/has_many.cr

```diff
@@ -1,7 +1,10 @@
 module Avram::Associations::HasMany
   macro has_many(type_declaration, through = nil, foreign_key = nil)
-    {% if !through.is_a?(NilLiteral) && !through.is_a?(SymbolLiteral) %}
-      {% through.raise "The association name for 'through' must be a Symbol. Instead, got: #{through}" %}
+    {% if !through.is_a?(NilLiteral) && (!through.is_a?(ArrayLiteral) || through.any? { |item| !item.is_a?(SymbolLiteral) }) %}
+      {% through.raise "'through' must be given an Array of Symbols. Instead, got: #{through}" %}
+    {% end %}
+    {% if !through.is_a?(NilLiteral) && through.size < 2 %}
```

Comment:
Do you expect this to work with more than 2? I tried a 3 level deep, but got errors.

```crystal
has_many things : Thing, through: [:first, :second, :things]
```

If that's not the case, then we should check that this is always exactly 2, and then make it a `Tuple(Symbol, Symbol)`. If we can allow more than 2, then we will need a whole test for that.

Reply:
right now it only handles two, but I plan to add multiple layers. I don't want to have multiple breaking changes though

Reply:
Ah ok. That's fine then